### DISCUSSION TITLE
Ethereum Payment Processing Configuration

### DISCUSSION THREAD

**Discussion by BitWizJason** [coinfoundry/miningcore#1034] [Q&A]
I am new to Ethereum and mining pools in general. I am trying to figure out what the best payment processing configuration is or at least better understand it.

The [example](https://github.com/oliverw/miningcore/blob/b353fdbe555af2e40209796cacbcaf405870cfdd/examples/ethereum_pool.json#L111) configuration in the repo shows:
```
"paymentProcessing": {
    "enabled": true,
    "minimumPayment": 1,
    "payoutScheme": "PPLNS",
    "payoutSchemeConfig": {
        "factor": 0.5
    },
    "gas": 21000,
    "maxFeePerGas": 50000000000,
    "keepUncles": false,
    "keepTransactionFees": false
}
```
1. For payoutSchemeConfig.factor the example sets it to .5. In the [code](https://github.com/oliverw/miningcore/blob/b353fdbe555af2e40209796cacbcaf405870cfdd/src/Miningcore/Payments/PaymentSchemes/PPLNSPaymentScheme.cs#L63) it defaults to 2 which I believe is what the [URL](https://bitcointalk.org/index.php?topic=39832) mentioned in the comment recommended. For what circumstances does a smaller value make more sense?
2.  Any thoughts on using [eth_estimateGas ](https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.estimate_gas) to automatically set the gas instead of hard coding it in the configuration?
3. Is the example value for maxFeePerGas a good average value or is there a good way to determine what this should be?
4. What happens if maxFeePerGas is set to low and the transaction gets stuck, dropped, or fails? Are the [two conditional exceptions](https://github.com/oliverw/miningcore/blob/b353fdbe555af2e40209796cacbcaf405870cfdd/src/Miningcore/Blockchain/Ethereum/EthereumPayoutHandler.cs#L410) meant to handle these scenarios?

Any help or advice is very appreciated!

**Comment by oliverw** [comment]
Some type of Gas fee optimization is already in the code base thanks to @Konstantin35. See 7a10969a5ee749c66a05b9542db597e47fd4846a

**Comment by Konstantin35** [comment]
Hello @BitWizJason , please read this and I think it will become clear to you how it works 
[maxpriorityfeepergas-vs-maxfeepergas](https://github.com/alchemyplatform/alchemy-docs/blob/master/guides/eip-1559/maxpriorityfeepergas-vs-maxfeepergas.md )

**Comment by BitWizJason** [comment]
Thank you @oliverw and @Konstantin35! Here is the [full page version](https://docs.alchemy.com/alchemy/guides/eip-1559/maxpriorityfeepergas-vs-maxfeepergas) of that same article in case others come across this discussion. I also found this guide [send-tx-eip-1559](https://docs.alchemy.com/alchemy/guides/eip-1559/send-tx-eip-1559) from alchemy which was very enlightening. I still have questions about some of the things I stated above but I will play around in the code, run some transactions and see what happens.
